src/Gridify.Elasticsearch/ToElasticsearchConverter.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text.Json;
+using Elastic.Clients.Elasticsearch;
+using Elastic.Clients.Elasticsearch.QueryDsl;
+using Gridify.Syntax;
+
+namespace Gridify.Elasticsearch;
+
+internal static class ToElasticsearchConverter
```

Comment:
Try to avoid duplicate code.

Reply:
I should note, I know how complicated this convertors are and also internal Gridify expression builder is not easy to read, so let me know if I can help on this or you had any questions.